inputs/org.osbuild.containers

```diff
@@ -32,6 +34,12 @@ SCHEMA = r"""
       "name": {
         "type": "string",
         "description": "The name to use for the image"
+      },
+      "format": {
+        "type": "string",
+        "enum": ["docker-archive", "oci-archive", "dir"],
+        "description": "The storage format of the container archive",
+        "default": "docker-archive"
```

Comment:
Specifically `docker-archive` is not a “source” format, in the sense that it’s not just a serialization of the registry data (that’s actually mostly what `dir:` is). Using `docker-archive` implies a _lossy_ conversion (decompress layers, drop the manifest; can’t represent multiple multi-arch variants).

(As an outsider, it also seems attractive to me to just hard-code one format instead of having to test and support three — but I wouldn’t know if there were any compatibility constraints e.g. with a version skew between consumers and producers.)

Reply:
I meant "source" in the **osbuild** terminology. Meaning a source that can be referenced by osbuild stage as an input. I used "source format" in this case to refer to the value of the "format" property of an osbuild stage input of type `org.osbuild.containers` referenced from the `source` section of an osbuild manifest.

The point I'm trying to make is that IMO, the producer (a source definition using a specific "format" for storing the container or an osbuild pipeline producing a container stored using a specific "format") predetermines the most sensible "format" to use for a stage  `org.osbuild.containers` input. I'm not saying that we should hardcode one specific "format". There is IMO just no point in allowing for it to be explicitly specified for the stage input, instead I suggest to make it implicit.

If I understand the PR and all comments correctly, it should be possible mostly to loose some information by specifying `docker-archive` or `oci-archive` as the format in a stage input, if the container was stored using `dir` format.

Therefore defaulting to the "format" in a stage `org.osbuild.containers` input that is used to store the container seems logical to me.

This does not imply that we can't test all "formats".